services/controllerhost/placement.go

```diff
@@ -77,7 +78,17 @@ func toResources(hosts []*common.HostInfo) []string {
 // Helper function to pick hosts based on the predicates
 func (p *DistancePlacement) pickHosts(service string, poolHosts, sourceHosts []*common.HostInfo, count int, minDistance, maxDistance uint16) ([]*common.HostInfo, error) {
 	if p.distMap == nil {
-		return poolHosts[:count], nil
+		if cnt := len(poolHosts); cnt >= count {
+			start := rand.Intn(cnt)
+			end := start + count
+			if end > cnt {
+				end = cnt
+			}
+			pickedHosts := append([]*common.HostInfo{}, poolHosts[start:end]...)
+			pickedHosts = append(pickedHosts, poolHosts[:count-len(pickedHosts)]...)
+			return pickedHosts, nil
```

Comment:
The above picks hosts from a random starting point, in a fixed order. It would probably be better to pick a truly random set of hosts:
```
pickedHosts := make([]*common.HostInfo, count)
for i, j := rand.Perm(len(poolHosts)) {
    if i >= count {
        break
    }
    pickedHosts[i] = poolHosts[j]
}
return pickedHosts, nil
```

Reply:
this is awesome..